src/main/java/com/aws/greengrass/shadowmanager/sync/CloudDataClient.java

```diff
@@ -283,7 +284,22 @@ private void handleUpdate(MqttMessage message) {
         String shadowName = shadowRequest.getShadowName();
         logger.atDebug().kv(LOG_THING_NAME_KEY, thingName).kv(LOG_SHADOW_NAME_KEY, shadowName)
                 .log("Received cloud update sync request");
-        syncHandler.pushLocalUpdateSyncRequest(thingName, shadowName, message.getPayload());
+        CompletableFuture
```

Comment:
Why are you using `CompletableFuture` here? Why not use `ExecutorService`?

Reply:
Mostly for ease of logging if an exception happens within the future (the `whenComplete()` call below), but it does run in context of ExecutorService: `.runAsync(() -> blah, executorService)`

Reply:
what exception can occur in `pushLocalUpdateSyncRequest`?

Is there a chance of local update requests being pushed out of order?

Reply:
> what exception can occur in `pushLocalUpdateSyncRequest`?

in LocalUdpateSyncRequest#isUpdateNecessary:

* `throw new SkipSyncRequestException(e);` for malformed shadow payload 
* new UnknownShadowException("Shadow not found in sync table") when searching in sync table


> Is there a chance of local update requests being pushed out of order?

presumably, since mqtt doesn't have ordering guarantees, or if we receive the message twice since it's QOS1